monty_profiling/floppy/flop_counting/operations/arithmetic.py

```diff
@@ -0,0 +1,71 @@
+import numpy as np
+import warnings
+from typing import Any, Optional, Tuple
+
+__all__ = ["Addition", "Subtraction", "Multiplication", "Division"]
+
+class ArithmeticOperation:
+    """Base class for arithmetic operations."""
+
+    def __init__(self, name: str):
+        self.name = name
+
+    def _compute_broadcast_shape(
+        self, *shapes: Tuple[int, ...]
+    ) -> Optional[Tuple[int, ...]]:
+        """Compute the broadcast shape for the inputs."""
+        try:
+            # Handle case where one input is scalar
+            if not all(shapes):
+                return shapes[1] if not shapes[0] else shapes[0]
+
+            # Create dummy arrays of ones with the given shapes
+            arrays = [np.ones(shape) for shape in shapes]
+            result = np.broadcast_arrays(*arrays)
+            return result[0].shape
+        except ValueError as e:
```

Comment:
Should we raise an error rather than a warning? Otherwise seems like it would be easy to overlook FLOPs that aren't being accounted for correctly.